### PR TITLE
LPS-132754 Test DB2 Upgrade

### PR DESCRIPTION
**Pull request by martamedio**
Hey @stian-sigvartsen 

@gaborlovasdotcom has tested our changes about SAML Upgrade Process and looks like it's still failing for DB2; so I again believe that the problem is the `min(samlSpSessionId) +` that we perform, I think we should try to to extract it.

Could you take a look to this? Can you check that the code I am running does the same as the previous `INSERT` would do? I'm afraid I'm missing something 😨 

Thanks!

### CODE REVIEW

**Comment by martamedio** [comment]
ci:test:security

**Comment by stian-sigvartsen** [comment]
@martamedio That code still has the problem we discussed that it will insert the same  `calculatedSamlSpSessionId` multiple times. When the SELECT part of the INSERT INTO returns more than one row. Let's take the discussion offline...

**Comment by martamedio** [comment]
ci:test:security

**Comment by martamedio** [comment]
ci:test:security

**Comment by martamedio** [comment]
ci:test:security

File: modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java [hunk 1]
```diff
@@ -38,35 +39,54 @@ protected void doUpgrade() throws Exception {
...
 
 			try (PreparedStatement preparedStatement =
 					connection.prepareStatement(
-						"select min(samlSpSessionId) - 1 from SamlSpSession");
+						StringBundler.concat(
+							"select min(samlIdpSpSessionId) as ",
+							"samlIdpSpSessionId, companyId, min(createDate) ",
+							"as createDate, userId, userName, nameIdFormat, ",
+							"nameIdNameQualifier, nameIdValue, samlSpEntityId ",
+							"from SamlIdpSpSession group by companyId, ",
```
Comment (stian-sigvartsen) [inline modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java:53]: This UpgradeProcess is for `SamlSpSession`. There is a separate one from `SamlIdpSpSession` which we can apply the same general approach to.

File: modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java [hunk 2]
```diff
@@ -38,35 +39,54 @@ protected void doUpgrade() throws Exception {
...
+						"nameIdNameQualifier");
+					String nameIdValue = resultSet.getString("nameIdValue");
+					String samlSpEntityId = resultSet.getString(
+						"samlSpEntityId");
+
+					int samlPeerBindingId =
+						samlIdpSpSessionId + -samlSpSessionIdOffset +
+							latestSamlPeerBindingId;
+
+					runSQL(
```
Comment (stian-sigvartsen) [inline modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java:76]: The runSQL() method will commit after each SQL statement, so if there is a `SQLException`, earlier SQL will re-run and cause duplicate entries in the `SamlPeerBinding` table. I think we can work around this by using its overloaded variety which accepts an existing `Connection`.

**Review by stian-sigvartsen — CHANGES_REQUESTED** [review]
General approach looks good I think 👍  
See individual comments.

File: modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java (hunk 1 above)
Comment (martamedio) [reply modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java]: OMG, I was reviewing the whole structure and I think I messed up with the copy & paste, good catch!
Fixed 😅

File: modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java (hunk 2 above)
Comment (martamedio) [reply modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java:75]: Done, take a look

**Comment by martamedio** [comment]
ci:test:security

**Comment by martamedio** [comment]
Please, see: https://github.com/liferay-appsec/liferay-portal/pull/398

**Pull request closed without merging**